front/packages/ui/src/browse/header.tsx

```diff
@@ -81,58 +113,91 @@ export const BrowseSettings = ({
+								}
+							/>
+						))}
+					</Menu>
+					<HR orientation="vertical" />
+					<IconButton
+						icon={GridView}
+						onPress={() => setLayout(Layout.Grid)}
+						color={layout === Layout.Grid ? theme.accent : undefined}
+						{...tooltip(t("browse.switchToGrid"))}
+						{...css({ padding: ts(0.5), marginY: "auto" })}
+					/>
+					<IconButton
+						icon={ViewList}
+						onPress={() => setLayout(Layout.List)}
+						color={layout === Layout.List ? theme.accent : undefined}
+						{...tooltip(t("browse.switchToList"))}
+						{...css({ padding: ts(0.5), marginY: "auto" })}
+					/>
+				</View>
 				<View {...css({ flexGrow: 1, flexDirection: "row", alignItems: "center" })}>
-					<Icon icon={Style} {...css({ marginX: ts(1) })} />
-					{filters.map((x) => (
-						<Chip key={x} label={x} />
-					))}
+					<Menu Trigger={MediaTypeTrigger} mediaType={mediaType}>
+						{availableMediaTypes.map((x) => (
+							<Menu.Item
+								key={x.key}
+								label={t(`browse.mediatypekey.${x.key}` as any)}
+								selected={mediaType === x}
+								icon={x.icon}
+								onSelect={() => {
+									if (mediaType === x || x === AllMediaTypes) {
+										setMediaType(undefined)
+									} else {
+										setMediaType(x)
+									}
+								}}
+							/>
+						))}
+					</Menu>
 				</View>
-			)}
-			<View {...css({ flexDirection: "row" })}>
-				<Menu Trigger={SortTrigger} sortKey={sortKey}>
-					{availableSorts.map((x) => (
-						<Menu.Item
-							key={x}
-							label={t(`browse.sortkey.${x}` as any)}
-							selected={sortKey === x}
-							icon={
-								x !== SearchSort.Relevance
-									? sortOrd === SortOrd.Asc
-										? ArrowUpward
-										: ArrowDownward
-									: undefined
-							}
-							onSelect={() =>
-								setSort(x, sortKey === x && sortOrd === SortOrd.Asc ? SortOrd.Desc : SortOrd.Asc)
-							}
-						/>
-					))}
-				</Menu>
-				<HR orientation="vertical" />
-				<IconButton
-					icon={GridView}
-					onPress={() => setLayout(Layout.Grid)}
-					color={layout === Layout.Grid ? theme.accent : undefined}
-					{...tooltip(t("browse.switchToGrid"))}
-					{...css({ padding: ts(0.5), marginY: "auto" })}
-				/>
-				<IconButton
-					icon={ViewList}
-					onPress={() => setLayout(Layout.List)}
-					color={layout === Layout.List ? theme.accent : undefined}
-					{...tooltip(t("browse.switchToList"))}
-					{...css({ padding: ts(0.5), marginY: "auto" })}
-				/>
 			</View>
-		</View>
+			<View
+				{...css({
+					flexDirection: "row-reverse",
+					alignItems: "center",
+					marginX: ts(4),
+					marginY: ts(1),
+					zIndex: 1,
+				})}
+			>
+				{filters.length !== 0 && (
+					<View {...css({ flexGrow: 1, flexDirection: "row", alignItems: "center" })}>
+						{/*<Icon icon={Style} {...css({ marginX: ts(1) })} />*/}
+						{filters.map((x) => (
+							<div style={{paddingRight: ".25rem"}}>
+								<Chip key={x} label={x} size={"small"} />
+							</div>
+						))}
+					</View>
+				)}
```

Comment:
This is never used, it was a WIP filter implementation. I think it can be removed with the new implementation.

Reply:
I was planning to introduce the filter chips once we have other filters (like genre etc) but you're right it shouldn't be in this PR. I'll remove it for now and we can add it back in if works well with other filters.